packages/components/src/templates/next/layouts/Collection/utils/__tests__/getCategoryFilter.test.ts

```diff
@@ -0,0 +1,56 @@
+import { describe, expect, it } from "vitest"
+
+import type { ProcessedCollectionCardProps } from "~/interfaces"
+import { getCategoryFilter } from "../getCategoryFilter"
+
+describe("getCategoryFilter", () => {
+  it("should return empty filter items when no items provided", () => {
+    // Arrange
+    const items: ProcessedCollectionCardProps[] = []
+
+    // Act
+    const result = getCategoryFilter(items)
+
+    // Assert
+    expect(result).toEqual({
+      id: "category",
+      label: "Category",
+      items: [],
+    })
+  })
+
+  it("should count and format categories correctly", () => {
+    // Arrange
+    const items: ProcessedCollectionCardProps[] = [
+      {
+        category: "guides",
+      } as ProcessedCollectionCardProps,
```

Comment:
issue: this type-cast is a little weird. I understand that we just want to test specific parts of the collection card props but this might hide problems if the implementation were to be changed, or if the props has changed upstream.

We can probably create a few complete test objects to be used across the different tests, or have some builder function that will automatically populate the remaining fields so that we can avoid this type casting.

Reply:
> might hide problems if the implementation were to be changed, or if the props has changed upstream.

hmm not quite understanding this, as this test should fail if the implementation changes (e.g. using more props), which should be the intended outcome of a test?